lib/modules/manager/terraform/extractors/others/modules.ts

```diff
@@ -60,7 +60,7 @@ export class ModuleExtractor extends DependencyExtractor {
     return dependencies;
   }
 
-  private analyseTerraformModule(dep: PackageDependency): PackageDependency {
+  public analyseTerraformModule(dep: PackageDependency): PackageDependency {
```

Comment:
```suggestion
  private analyseTerraformModule(dep: PackageDependency): PackageDependency {
```

Please test your changes end-to-end, i.e. in the `extract.spec.ts` with the simplest-case input.

I checked this code in the browser console, and it didn't trim the query part:
```js
`git::https://example.gitlab-dedicated.com/group/subgroup/repo.git?ref=v2.0.0`.replace(/\.git(?=$|[/?])/g, '')
```